src/js/options.js

```diff
@@ -298,31 +321,93 @@ var SettingsTab = {
     }
   },
 
-  saveImport: async function () {
+  saveSearches: async function (data) {
     try {
-      var parsedImport = JSON.parse(
-        document.getElementById("settings_json").value
-      );
+      var parsedData = JSON.parse(data);
 
       if (
         confirm(
           "Are you sure you want to override your local settings with these values?"
         )
       ) {
-        await ConfigFile.parseJSONFile(parsedImport, true);
+        await ConfigFile.parseJSONFile(parsedData, true);
 
         // Update UI according to this change
         mainConfigurationUpdated();
 
         Notiflix.Notify.Success("New configuration saved");
+        return true;
       }
     } catch (err) {
       console.error(err);
       Notiflix.Notify.Failure(
-        "Configuration couldn't be saved. Please verify that the file is valid."
+        "Configuration couldn't be saved. Please verify that the file/data is valid."
       );
+      return false;
+    }
+  },
+
+  importFromFile() {
+    document.getElementById("settings_fileInput").click();
+  },
+
+  fileImported(event) {
+    let selectedFile = _.first(this.files);
+    var reader = new FileReader();
+
+    reader.onload = function (event) {
+      SettingsTab.saveSearches(event.target.result);
+    };
+    reader.readAsText(selectedFile);
+
+    // Reset the value
+    event.target.value = "";
+  },
+
+  exportToFile: async function () {
+    // Generate the JSON
+    let data = await ConfigFile.generateJSONFile();
+
+    // Create a downloadable link with content. I have named the exported file to `Settings.json`
+    var downloadLink = document.createElement("a");
+    var blob = new Blob([JSON.stringify(data)], {
+      type: "text/plain;charset=utf-8",
+    });
+    var url = URL.createObjectURL(blob);
+    downloadLink.href = url;
+    downloadLink.download = exportFileName;
+
+    document.body.appendChild(downloadLink);
+    downloadLink.click();
+    document.body.removeChild(downloadLink);
+  },
+
+  openModal: async function () {
+    // Update text Area
+    await SettingsTab.updateJSONTextarea();
+
+    document.getElementById("Settings_editModalBackdrop").style.display =
+      "block";
+    document.getElementById("settings_editModal").style.display = "block";
+    document.getElementById("settings_editModal").classList.add("show");
+  },
+
+  saveModalChanges() {
+    let changes = document.getElementById("settings_json").value;
+    var success = SettingsTab.saveSearches(changes);
+
+    // close Modal if saved
+    if (success) {
+      SettingsTab.closeModal();
```

Comment:
I tried entering an invalid value on the textarea, but the modal is closed anyway